src/main/resources/library/attribute-definitions.json

```diff
@@ -7,7 +7,7 @@
   "required": ["library:datasetName", "library:datasetVersion", "library:datasetDescription", "library:datasetCustodian",
     "library:datasetDepositor", "library:contactEmail", "library:datasetOwner", "library:institute", "library:indication",
     "library:numSubjects", "library:projectName", "library:dataCategory", "library:datatype", "library:dataUseRestriction",
-    "library:studyDesign", "library:cellType", "library:requiresExternalApproval"],
+    "library:studyDesign", "library:cellType", "library:requiresExternalApproval", "library:useLimitationOption"],
```

Comment:
if you change the `oneOf` just below this to:

```
  "oneOf": [{
    "required": ["library:GRU", "library:HMB", "library:NCU", "library:NPU", "library:NDMS", "library:NAGR",
      "library:NCTRL", "library:RS-G", "library:RS-PD", "library:useLimitationOption"],
    "properties": {
      "library:useLimitationOption": {
        "type": "string",
        "enum": ["questionnaire"]
        }
      }
  }, {
    "required": ["library:orsp", "library:useLimitationOption"],
    "properties": {
      "library:useLimitationOption": {
        "type": "string",
        "enum": ["orsp"]
      }
    }
  }],
```

then I think you get the appropriate conditional validation in the API. Essentially, you're requiring a specific value of library:useLimitationOption for each choice in the `oneOf`. 

Note that you do NOT get good schema validation messages for those violations - we'd have to tweak that further. And, because we expect all props to exist in the top-level `properties` key, this creates some duplication. But it seems like a good path to explore.